client/src/components/ExpandedRowComponent.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {MouseEvent} from 'react';
+import type {CodeReview} from '../lib/types';
+import {format} from 'date-fns';
+import {logError} from '../services/log';
+import {useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+export const useExpandedRowComponent = ({onUpdate, user}: {onUpdate: CallableFunction; user?: {displayName: string}}) => {
+    const ExpandedRowComponent = ({data}: {data: CodeReview}) => {
+        const handleActionClick = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+            const action = currentTarget.getAttribute('name');
+            const value = currentTarget.getAttribute('value');
+
+            fetchData(`/api/action/${action}/${value}`)
+                .then((result) => {
+                    result?.data && onUpdate && onUpdate({
```

Comment:
onUpdate should be a required prop and then this wouldn't have to be checked here.